### PR TITLE
feat: 🎸 added signing and verifying of data of webatuhn token

### PR DESCRIPTION
**Pull request by xpavlic**
Added signing and verifying of data of webatuhn tokens stored in remote
storage. If token data sign is verified and logged user and token
user_ids are same, token can be used to authentication. Added PerunStorage logger and
loggerPrefix enum.

### CODE REVIEW

File: app/perunStorage.py [hunk 1]
```diff
@@ -58,18 +60,38 @@ def _ldap_search_webauthn_tokens(self, username):
...
+    def _verify_signed_data(self, data):
+        key = self._get_jwk()
+        jws_token = jws.JWS()
+        jws_token.deserialize(data)
+        try:
+            jws_token.verify(key)
+        except InvalidJWSSignature:
+            return False
+        return json.loads(jws_token.payload.decode('utf-8'))
+
```
Comment (melanger) [inline app/perunStorage.py:86]: Can it be backward compatible? How about adding a configuration option (bool) signed_tokens (default false), if false keep original behavior = _get_signed_data just returns data and _verify_signed_data just returns data; and if true then do the new behavior.

File: app/perunStorage.py [hunk 2]
```diff
@@ -58,18 +60,38 @@ def _ldap_search_webauthn_tokens(self, username):
         result = []
         for token in tokens:
             token = json.loads(token)
-            if not token["revoked"] and token["type"] == 'WebAuthn':
-                result.append(token)
+            if not token['revoked'] and token['type'] == 'WebAuthn':
+                data = self._verify_signed_data(token['data'])
+                if data['username'] == username:
```
Comment (melanger) [inline app/perunStorage.py:65]: e.g. `if not config['signed_tokens'] or data['username'] == username:`

**Review by melanger — CHANGES_REQUESTED** [review]
Try to make it backward-compatible.
Also, check what happens when users has no tokens in LDAP or he has only "faked" tokens, e.g. no tokens with correct signature and/or username.
Username mismatch and signature mismatch should be at least logged, this means a configuration error or a security incident.

**Comment by xpavlic** [comment]
@melanger if user has only "faked" toknes, the behavior is the same as behavior for user without tokens, so the app redirects him to the registration page.

**Review by melanger — APPROVED** [review]

**Pull request merged**